hack/tools/go.mod

```diff
@@ -9,6 +9,7 @@ replace sigs.k8s.io/cluster-api/test => ../../test
 require (
 	cloud.google.com/go/storage v1.35.1
 	github.com/blang/semver/v4 v4.0.0
+	github.com/onsi/gomega v1.30.0
```

Comment:
This is my only real change
The other modules just come from running `go mod tidy`